codegen/src/api/mod.rs

```diff
@@ -81,7 +81,7 @@ impl RuntimeGenerator {
 
         Ok(quote! {
             #( #item_mod_attrs )*
-            #[allow(dead_code, unused_imports, non_camel_case_types)]
+            #[allow(dead_code, unused_imports, non_camel_case_types, unreachable_patterns)]
```

Comment:
This is not really a clippy lint but a rustc warning.

I don't mind adding this but more interesting why/when the codegen would generate something like empty enums...

Reply:
@niklasad1 e.g. you can generate it for rococo/westend and run `cargo +nighlty clippy`, e.g.:
https://github.com/paritytech/parity-bridges-common/pull/3059/files#diff-611bdbe35fbbfe14f550dc53edae00c57c91bbd07b6fc7ad3cdec38f7f2c6cdb

```
git clone git@github.com:paritytech/parity-bridges-common.git
cd parity-bridges-common/tools/runtime-codegen

cargo run --bin runtime-codegen -- --from-node-url "wss://rococo-rpc.polkadot.io:443" > ../../relay-clients/client-rococo/src/codegen_runtime.rs

cargo run --bin runtime-codegen -- --from-node-url "wss://westend-rpc.polkadot.io:443" > ../../relay-clients/client-westend/src/codegen_runtime.rs
```